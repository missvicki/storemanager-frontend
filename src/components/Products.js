import React from 'react';
import { Header, Table, Container } from 'semantic-ui-react';


const renderProducts = (data) => {
    const productArray = data.product;
    if (productArray) {
        return productArray.map(product => (
            <Table.Row key={product.product_id}>
                <Table.Cell>{product.product_name}</Table.Cell>
                <Table.Cell>{product.category}</Table.Cell>
                <Table.Cell>{product.unit_price}</Table.Cell>
                <Table.Cell>{product.quantity}</Table.Cell>
            </Table.Row>
        ));
    }
};

const Products = props => {
    return (
        <div>
            <Container>
                <br />
                <Header as="h1" textAlign="center" padded="very" style={{ color: "gray" }}>Products</Header>
                <Table unstackable>
                    <Table.Header>
                        <Table.HeaderCell>Product Name</Table.HeaderCell>
                        <Table.HeaderCell>Category</Table.HeaderCell>
                        <Table.HeaderCell>Unit Price</Table.HeaderCell>
                        <Table.HeaderCell>Quantity</Table.HeaderCell>
                    </Table.Header>
                    <Table.Body>
                        {renderProducts(props)}
                    </Table.Body>
                </Table>
            </Container>
        </div>
    );
};

export default Products;
