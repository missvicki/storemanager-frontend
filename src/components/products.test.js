import React from 'react';
import { shallow } from 'enzyme';

import Products from './Products';

const props = {
    products: []
};

const products = shallow(<Products {...props}/>);

describe('Products', () => {
    it('should render the Product Component', () => {
        expect(products).toMatchSnapshot();
    });
});
