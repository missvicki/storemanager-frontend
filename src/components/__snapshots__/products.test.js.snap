// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products should render the Product Component 1`] = `
<div>
  <Container>
    <br />
    <Header
      as="h1"
      padded="very"
      style={
        Object {
          "color": "gray",
        }
      }
      textAlign="center"
    >
      Products
    </Header>
    <Table
      as="table"
      unstackable={true}
    >
      <TableHeader
        as="thead"
      >
        <TableHeaderCell
          as="th"
        >
          Product Name
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Category
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Unit Price
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Quantity
        </TableHeaderCell>
      </TableHeader>
      <TableBody
        as="tbody"
      />
    </Table>
  </Container>
</div>
`;
